x/upgrade/keeper/keeper.go

```diff
@@ -400,11 +397,11 @@ func (k Keeper) GetUpgradePlan(ctx context.Context) (plan types.Plan, err error)
 
 // setDone marks this upgrade name as being done so the name can't be reused accidentally
 func (k Keeper) setDone(ctx context.Context, name string) error {
-	store := k.storeService.OpenKVStore(ctx)
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	k.Logger(ctx).Debug("setting done", "height", sdkCtx.HeaderInfo().Height, "name", name)
+	store := k.environment.KVStoreService.OpenKVStore(ctx)
 
-	return store.Set(encodeDoneKey(name, sdkCtx.HeaderInfo().Height), []byte{1})
+	k.Logger(ctx).Debug("setting done", "height", k.environment.HeaderService.GetHeaderInfo(ctx).Height, "name", name)
+
+	return store.Set(encodeDoneKey(name, k.environment.HeaderService.GetHeaderInfo(ctx).Height), []byte{1})
```

Comment:
Unrelated, I didn't know upgrade didn't use collections. So we missed this one next to https://github.com/cosmos/cosmos-sdk/issues/18280 and https://github.com/cosmos/cosmos-sdk/issues/18456